### PR TITLE
NO_VGA fixes (for bitbox)

### PR DESCRIPTION
**Pull request by lowagner**
should fix the issue i just raised, at least for certain configurations.

### CODE REVIEW

**Comment by lowagner** [comment]
i am a bit "worried" by commenting this out in the bitbox_main file:
```
    int oframe=vga_frame;
    while (oframe==vga_frame);
```

maybe there's a way with NO_VGA to still have vga_frame keeping time, but just without sending pixels out to VGA...  not sure if it's horrible to call game_frame constantly, but it might be nice to have a small sleep session??

**Comment by makapuf** [comment]
what could be one is that when we have NOVGA, we generally have sound, so we may sync with sound buffers ... Other possibility, we keep vsync at 60Hz but disable DMA output and line rendering callbacks.  Third possibility, NOVGA doesn't exist and you just make an empty graph_line and let standard mode going. It's indeed a bit strange to have a game_frame with no vga.

**Comment by lowagner** [comment]
that sounds fine, and i have similar preferences.  though maybe this is a 4.5:

i mainly figure a NO_VGA option should reduce the extra computing requirements, allowing one to focus on sound.  does the game_snd_buffer get called once a "frame" so we could maybe set vga_frame to update when that is called?  perhaps it's a bit messy, so it should be left to the end user.

**Comment by makapuf** [comment]
> does the game_snd_buffer get called once a "frame" so we could maybe set vga_frame to update when that is called?
no, they are on independent timers so that buffer size is not linked to the number of lines or vsync time (i.e different for each). But if you're setting NOVGA, maybe you shouldn't care about video timing and more about sound timing (and then create yourself a static buffer_count in your sound callback). 

I'd suggest 4 in fact

**Comment by lowagner** [comment]
ok, so this "buffer_count" timer would then depend on the sample rate and the buffer length, correct?

it still might be nice to have a "game_frame" that gets called as often as possible, in case somebody is thinking of a no-audio option to calculate something (to put to the SD card), or something else.  (option 0, which is my pull request currently...) otherwise they'd have to define a new main, which is doable, but maybe not as intuitive as just keeping a game_frame.

**Comment by makapuf** [comment]
> this "buffer_count" timer would then depend on the sample rate and the buffer length, correct?

well it will be called in sync with audio, every 512 samples by example. so yes, frequency would depend on sample rate and buffer len.

> in case somebody is thinking of a no-audio option to calculate something (to put to the SD card), or something else

well I'd say in that case we only provide a game_init in which the user can do anything, including a `while(1) compute()` - in this case game_init is not much different than main

but the case where we have neither audio (for which snd callback will be the basis for anything) nor vga has for me only one case : led_blink :)  

so basically, 0 or 4 are fine for me. if you prefer it I can apply your patch as is.

**Comment by lowagner** [comment]
as you wish, i am fine with anything.
